lib/ansible/modules/network/edgeos/edgeos_config.py

```diff
@@ -186,15 +186,34 @@ def get_candidate(module):
     return config_to_commands(contents)
 
 
-def diff_config(commands, config):
-    config = [to_native(c).replace("'", '') for c in config.splitlines()]
+def check_command(module, command):
+    """Tests against a command line to be valid otherwise raise errors
+
+    Error on uneven single quote which breaks ansible waiting for further input. Ansible will handle
+    even single quote failures correctly.
+
+    :param command: the command line from current or new config
+    :type command: string
+    :raises ValueError:
+      * if contains odd number of single quotes
+    :return: command string unchanged
+    :rtype: string
+    """
+    if command.count("'") % 2 != 0:
+        module.fail_json(msg="Ansible does not support single (') quotes in values: " + command)
```

Comment:
I think this message is a bit confusing. It sounds like single quotes are not allowed in commands at all, which is not the case. Unmatch or uneven quotes are the problem.
```suggestion
        module.fail_json(msg="Unmatched single (') quote found in command: " + command)
```

Reply:
agreed and updated